### PR TITLE
refactor: allow using instant-seal on all runtimes

### PR DESCRIPTION
**Pull request by gregdhill**
Closes https://github.com/interlay/interbtc/issues/955

### CODE REVIEW

File: parachain/runtime/common/src/lib.rs [hunk 1]
```diff
@@ -113,3 +112,19 @@ where
...
+
+pub struct MaybeSetTimestamp<T>(PhantomData<T>);
+
+impl<T> OnTimestampSet<T::Moment> for MaybeSetTimestamp<T>
+where
+    T: frame_system::Config + pallet_aura::Config + pallet_sudo::Config,
+{
+	fn on_timestamp_set(moment: T::Moment) {
+        // key is not set on mainnet
+        if pallet_sudo::Pallet::<T>::key().is_none() {
```
Comment (gregdhill) [inline parachain/runtime/common/src/lib.rs:124]: I'm not sure if this is the best approach but I couldn't see an obvious alternative. @sander2 what do you think?

Comment (sander2) [reply parachain/runtime/common/src/lib.rs:124]: A bit hacky for sure but without knowing why exactly the hook fails I can't really suggest better alternatives. I guess it's fine

File: parachain/src/chain_spec/testnet_interlay.rs [hunk 2]
```diff
@@ -65,14 +65,14 @@ pub fn staging_mainnet_config(benchmarking: bool) -> InterlayChainSpec {
...
+                // 5Ec37KSdjSbGKoQN4evLXrZskjc7jxXYrowPHEtH2MzRC7mv (//sudo/1)
+                Some(get_account_id_from_string(
+                    "5Ec37KSdjSbGKoQN4evLXrZskjc7jxXYrowPHEtH2MzRC7mv",
+                )),
                 crate::chain_spec::interlay::PARA_ID.into(),
                 DEFAULT_BITCOIN_CONFIRMATIONS,
             );
 
-            // 5Ec37KSdjSbGKoQN4evLXrZskjc7jxXYrowPHEtH2MzRC7mv (//sudo/1)
-            genesis.sudo.key = Some(get_account_id_from_string(
```
Comment (sander2) [inline parachain/src/chain_spec/testnet_interlay.rs:73]: not sure why you refactored this but kept the btc_relay config override unchanged, but I guess it doesn't matter a ton

**Review by sander2 — APPROVED** [review]

File: parachain/runtime/common/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply parachain/runtime/common/src/lib.rs:124]: This is the error: https://github.com/paritytech/substrate/blob/83bf140ca43df76ccaf422385562e33782ed43f8/frame/timestamp/src/lib.rs#L176

Comment (gregdhill) [reply parachain/runtime/common/src/lib.rs:124]: Which causes the panic here: https://github.com/paritytech/substrate/blob/83bf140ca43df76ccaf422385562e33782ed43f8/frame/timestamp/src/lib.rs#L176

**Pull request merged**